### ISSUE TITLE
Leverage metavariable matching in sgrep-lint

### ISSUE THREAD

**Issue by nbrahms** [semgrep/semgrep#405]
#403 added metavariable IDs to track variables between patterns. Let's use this in sgrep-lint.

**Comment by brendongo** [comment]
I'm assuming this is referring to https://github.com/returntocorp/sgrep/pull/403/files

**Comment by DrewDennison** [comment]
#403 is merged so this is unblocked

**Comment by aryx** [comment]
See https://github.com/returntocorp/sgrep/issues/434 as an example which require this feature.

**Comment by dlukeomalley** [comment]
Is this a duplicate of #176?
